_source/_posts/2020-12-04-25-years-javascript-java.adoc

```diff
@@ -0,0 +1,156 @@
+tweets:
+- ""
+- ""
+- ""
+image:
+type: awareness
+---
+:toc: macro
+:page-liquid:
+:experimental:
+
+The Year is 1995...
+
+Java is born in May. So is JavaScript (but it's called Mocha).
+
+Windows 95 is released in August.
+
+It's widely circulated that JavaScript first appeared on this day, December 4th.
+
+Wow! It's so crazy to look back and see so many influential software releases happen in such a short period.
+////
+It's almost like when Flex, JSF, and Spring were all released in March 2004. 😜
+////
+
+
+Congrats to both JavaScript and Java for doing so well over the last 25 years!
+
+toc::[]
+
+== A Brief History of JavaScript
+
+JavaScript went through quite the transformation in 1995. It started as an idea for a "glue language" for web designers and hobbyists, while Java was the "component language" for higher-priced programmers.
+
+According to Dr. Axel Rauschmayer's "Speaking JavaScript" book, specifically http://speakingjs.com/es5/ch04.html[How JavaScript Was Created]:
+
+> Eich wrote one [scripting language] in 10 days, in May 1995. JavaScript's first code name was Mocha, coined by Marc Andreesen. Netscape marketing later changed it to LiveScript, for trademark reasons and because the names of several products already had the prefix "Live." In late November 1995, Navigator 2.0B3 came out and included the prototype, which continued its early existence without major changes. In early December 1995, Java's momentum had grown and the language was renamed again, to its final name, JavaScript.
+
+Yes, it's true. JavaScript started as a marketing ploy. The higher-ups thought it'd be smart if it was positioned as a _companion_ to Java. 25 years later, I think https://jhipster.tech[they're excellent companions]!
+
+The JavaScript Renaissance started in the mid-2000s with jQuery. jQuery leveled the playing field so developers could write code that would work on most popular browsers, and freed developers of the browser wars burden. From there, full-fledged JavaScript frameworks like Ember.js, Backbone.js, and AngularJS took the developer world by storm.
+
+image::{% asset_path 'blog/25-years-javascript-java/jquery.png' %}[alt=jQuery: write less, do more,width=400,align=center]
+
+In 2009, https://nodejs.dev/learn/a-brief-history-of-nodejs[Node.js was created], as well as the first version of `npm`. Node allows you to run JavaScript outside of a browser and JavaScript found itself well-positioned on the server.
+
+image::{% asset_path 'blog/25-years-javascript-java/node.png' %}[alt=Node.js: designed to build scalable network applications,align=center,width=400]
+
+In 2013, GitHub created Electron and introduced a way to write multi-platform desktop applications with JavaScript. Some of the most often-used software developers use today (Slack and VS Code) are powered by Electron!
+
+In 2014++, along came React, Angular (with TypeScript), and Vue. They continue to be the dominant frameworks for writing browser-based applications.
```

Comment:
`++` ?

Reply:
I meant to imply "in 2014 and subsequent years". Maybe there's a better way to express this?